contributing/service-discovery/dns-matrix.md

```diff
@@ -0,0 +1,12 @@
+|                | service                 | connect (enterprise)    | ingress (enterprise)    | node                   | query           | addr                         |
```

Comment:
A quick intro paragraph would be great to explain what this is, and maybe also to say that only a single `OpCode` is supported.